x/ccv/parent/parent_test.go

```diff
@@ -722,3 +763,157 @@ func (s *ParentTestSuite) UpdateChildHistInfo(changes []abci.ValidatorUpdate) {
+	pApp.MM.OrderBeginBlockers[i] = distrtypes.ModuleName // set value
+}
+
+// TestDistribution tests that tokens are distributed to the
+// provider chain from the consumer chain appropriately
+func (s *ParentTestSuite) TestDistribution() {
+	s.SetupCCVChannel() // also sets up transfer channels
+	// NOTE s.transferPath.EndpointA == Consumer Chain
+	//      s.transferPath.EndpointB == Provider Chain
+
+	pChain, cChain := s.parentChain, s.childChain
+	pApp, cApp := pChain.App.(*app.App), cChain.App.(*app.App)
+	cKeep := cApp.ChildKeeper
+
+	// Get the receiving fee pool on the provider chain
+	fcAddr := pApp.ParentKeeper.GetFeeCollectorAddressStr(pChain.GetContext())
+
+	// Ensure that the provider fee pool address stored on the consumer chain
+	// is the correct address
+	fcAddr2 := cApp.ChildKeeper.GetProviderFeePoolAddrStr(cChain.GetContext())
+	s.Require().Equal(fcAddr, fcAddr2)
+
+	// make sure we're starting at consumer height 21 (some blocks commited during setup)
+	s.Require().Equal(int64(21), cChain.GetContext().BlockHeight())
+
+	// get last consumer transmission
+	ltbh, err := cKeep.GetLastTransmissionBlockHeight(cChain.GetContext())
+	s.Require().NoError(err)
+	s.Require().Equal(int64(0), ltbh.Height)
+
+	bpdt := cKeep.GetBlocksPerDistributionTransmission(cChain.GetContext())
+	s.Require().Equal(int64(1000), bpdt)
+
+	// check the consumer chain fee pool
+	consumerFeePoolAddr := cApp.AccountKeeper.GetModuleAccount(
+		cChain.GetContext(), authtypes.FeeCollectorName).GetAddress()
+	providerFeePoolAddr := pApp.AccountKeeper.GetModuleAccount(
+		pChain.GetContext(), authtypes.FeeCollectorName).GetAddress()
+	balance := cApp.BankKeeper.GetBalance(cChain.GetContext(), consumerFeePoolAddr, "stake")
+	s.Assert().Equal(balance.Amount.Int64(), int64(140062235461521))
+
+	// Commit some new blocks (commit blocks less than the distribution event blocks)
+	s.coordinator.CommitNBlocks(cChain, (1000-1)-21)
+	err = s.path.EndpointB.UpdateClient()
+	s.Require().Equal(int64(1000), cChain.GetContext().BlockHeight())
+
+	// check the consumer chain fee pool (should have increased
+	balance = cApp.BankKeeper.GetBalance(cChain.GetContext(), consumerFeePoolAddr, "stake")
+	expFeePoolAtDistr := int64(4175822659438993)
+	s.Assert().Equal(balance.Amount.Int64(), expFeePoolAtDistr)
+
+	// Verify that the destinationChannel exists
+	// if this doesn't exist then the transfer logic will fail when
+	// a the distribution transfer is invoked in the next block.
+	ctx := cChain.GetContext()
+	sourcePort := transfertypes.PortID
+	sourceChannel := cKeep.GetDistributionTransmissionChannel(ctx)
+	sourceChannelEnd, found := cApp.IBCKeeper.ChannelKeeper.GetChannel(ctx, sourcePort, sourceChannel)
+	s.Require().True(found)
+	destinationChannel := sourceChannelEnd.GetCounterparty().GetChannelID()
+	s.Require().True(len(destinationChannel) > 0)
+
+	// commit 1 more block (which should invoke a distribution event)
+	rspEB, _, _ := s.coordinator.CommitBlockGetResponses(cChain)
```

Comment:
Nice we should upstream any event parsing you've added to the test suite

Reply:
Yeah, I've added this to a ibc branch already... will open a PR before resolving this conversion

Reply:
Alright - here is the PR (still in DNM status)... needs to be rebased looks like, still good to have this PR open for reference purposes. https://github.com/cosmos/ibc-go/pull/1304